### PR TITLE
update `genesis_config` related docs and tests and error messages

### PR DESCRIPTION
**Pull request by kianenigma**
Follow-up to https://github.com/paritytech/substrate/pull/14306.

I hope this also showcases the important message of: **It is really not that hard to make the examples codes in rust-docs compile, and therefore remain correct. Please embrace this :)** 

It moves the documentation of proc macros to their re-export, such that can link other items in frame-support. This is a patter that we should embrace for all of macro docs, and apply in PRs like https://github.com/paritytech/substrate/pull/13987 as well.

### CODE REVIEW

File: substrate/frame/support/src/lib.rs [hunk 1]
```diff
@@ -2188,10 +2188,80 @@ pub mod pallet_macros {
...
+
+	/// Allows you to define the genesis configuration for the pallet.
+	///
+	/// Item is defined as either an enum or a struct. It needs to be public and implement the
+	/// trait [`frame_support::traits::BuildGenesisConfig`].
+	///
+	/// See [`genesis_build`] for an example.
+	pub use frame_support_procedural::genesis_config;
+
+	/// Allows you to define how the state of your pallet at genesis is build is built. This
```
Comment (gpestana) [inline substrate/frame/support/src/lib.rs:2204]:
```suggestion
	/// Allows you to define how the state of your pallet at genesis is built. This
```

**Review by gpestana — APPROVED** [review]

**Review by wentelteefje — APPROVED** [review]

**Comment by sam0x17** [comment]
This is fine, but keep in mind if doing this technique, unless something has changed since the last time I checked, rust analyzer will not pick up on the re-export docs when hovering over those items, so it might make sense to put the "main docs" for the thing on the actual item, and the "examples" on the re-export. I think rustdoc will put one before the other and I forget which order but worth playing around with to see if we can get _some_ docs in RA and all the docs in everything else.

This was the reason for the documented macro stubs that I added

**Comment by kianenigma** [comment]
> I think rustdoc will put one before the other and I forget which order but worth playing around with to see if we can get some docs in RA and all the docs in everything else instead of no docs in RA because that defeats the purpose of the stubs.

The one on the re-export goes first, so putting the example first is a weird experience.. 

Thanks for pointing this out. I would have to find another alternative. This is still a pretty much unsolved issue. Why is it that RA will get the stubs first, and not the re-exports?

**Review by Rashmirreddy — APPROVED** [review]

**Comment by ggwpez** [comment]
bot update-ui  --rust_version=1.70

**Comment by ggwpez** [comment]
bot update-ui substrate --rust_version=1.70

**Review by franciscoaguirre — APPROVED** [review]
Love functioning doc tests 😍

**Review by juangirini — APPROVED** [review]

**Review by sam0x17 — APPROVED** [review]

**Review by michalkucharczyk — APPROVED** [review]

**Comment by kianenigma** [comment]
I was not done lol :D

**Comment by kianenigma** [comment]
In retrospect, this was part of https://github.com/paritytech/polkadot-sdk/issues/247

**Pull request merged**